### PR TITLE
Machine Learning with XGBoost and Scikit-learn

### LINKED ISSUE #3966: [Machine Learning]Machine XGBoost and Sklearn Python from scratch
#### Proposed title of article
[Machine Learning]Machine Learning with XGBoost and Sklearn Python from scratch

Use keyword research to improve your article's SEO.

#### Proposed article introduction
*We're looking for the first 2-3 paragraphs of your article that appropriately summarize what your article will be about.*
We will use this writing snippet to help us assess the overall quality before approval so ensure it has been proofread, spellchecked and grammarchecked.

For example, if the article is based on machine learning, use the following: [Machine learning] Introduction to Machine Learning.
If the article is based on developing an Android application, use the following: [Android] Developing Apps using Android.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?
Use this area to get your ideas down on the bulk of your article or tutorial.

#### Article quality
How is your article unique? Tell us what makes your approach different from similar articles that have published on the same topic?
Is yours more in-depth? Does it cover additional topics? Do you provide handy tips or anecdotal advice?

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Bravinsimiyu**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [ ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [ ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [ ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [ ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [ ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [ ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ ] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [ ] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [ ] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3966 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+
+### Prerequisites
+
+A reader must have:
+
+1. A good understanding of [Python](/engineering-education/python-projects-for-beginners/).
+2. A good understanding of [machine learning modeling](/engineering-education/house-price-prediction/).
+3. A good understanding of [supervised learning algorithms](/engineering-education/supervised-learning-algorithms/)
+4. Have knowledge of [Pandas](https://numpy.org/) and [Numpy.](https://numpy.org/)
+   > NOTE: To follow along easily, use [Google Colab](https://research.google.com/) to build your model.
```
Comment (srishilesh) [inline content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md:33]:
```suggestion

> NOTE: To follow along easily, use [Google Colab](https://research.google.com/) to build your model.
```

File: content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+- [References](#references)
+
+### Prerequisites
+
+A reader must have:
+
+1. A good understanding of [Python](/engineering-education/python-projects-for-beginners/).
+2. A good understanding of [machine learning modeling](/engineering-education/house-price-prediction/).
+3. A good understanding of [supervised learning algorithms](/engineering-education/supervised-learning-algorithms/)
+4. Have knowledge of [Pandas](https://numpy.org/) and [Numpy.](https://numpy.org/)
```
Comment (srishilesh) [inline content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md:32]:
```suggestion
- A good understanding of [Python](/engineering-education/python-projects-for-beginners/).
- A good understanding of [machine learning modeling](/engineering-education/house-price-prediction/).
- A good understanding of [supervised learning algorithms](/engineering-education/supervised-learning-algorithms/)
- Have knowledge of [Pandas](https://numpy.org/) and [Numpy.](https://numpy.org/)
```

File: content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)
+```
+
+To understand the power of XGBoost we need to compare it with another algorithm, we first use the decision tree classifier algorithm to build the model. After that, we then build the same model using XGBoost and compare the results to see if XGBoost has improved the model performance.
+
+We start with a decision tree classifier.
+
+### Decision tree classifier
+
+A decision tree classifier is a machine learning algorithm that is used to solve classification problems. It's imported from the Scikit-learn Python library.
```
Comment (srishilesh) [inline content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md:233]: It would be great if you give an insight on how it works.

Using inbuilt scikit library is easier by abstracting the working, but understanding it's working is essential.

File: content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+Accuracy of Model:: 0.8929351784413693
+
+```
+
+This shows that the model has an accuracy score of `89.29%` when making predictions.
+
+Let's see if XGBoost can improve the performance of this model and increase the accuracy score.
+
+### XGBoost
+
```
Comment (srishilesh) [inline content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md:312]: Explain in-detail about XGBoost and its working.

File: content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+```bash
+
+xgb_classifier.fit(X_train,y_train)
+XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
+       colsample_bynode=1, colsample_bytree=1, gamma=0, learning_rate=0.1,
+       max_delta_step=0, max_depth=3, min_child_weight=1, missing=None,
+       n_estimators=100, n_jobs=1, nthread=None,
+       objective='binary:logistic', random_state=0, reg_alpha=0,
+       reg_lambda=1, scale_pos_weight=1, seed=None, silent=None,
+       subsample=1, verbosity=1)
```
Comment (srishilesh) [inline content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md:342]: Explain these parameters on why they are used

File: content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+
+After training this is the output.
+
+```bash
+DecisionTreeClassifier(class_weight=None, criterion='gini', max_depth=None,
+            max_features=None, max_leaf_nodes=None,
+            min_impurity_decrease=0.0, min_impurity_split=None,
+            min_samples_leaf=1, min_samples_split=2,
+            min_weight_fraction_leaf=0.0, presort=False, random_state=None,
+            splitter='best')
```
Comment (srishilesh) [inline content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md:263]: Explain the parameters

**Review by srishilesh — CHANGES_REQUESTED** [review]
@Bravinsimiyu 

Great article.

The formatting looks good, but content-wise I've suggested a few changes.

Please let me know when it's done.

Thanks.

**Comment by LinusMuema** [comment]
Hi @Bravinsimiyu 

Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Machine Learning with XGBoost and Scikit-learn #4113 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7283635/Machine.Learning.with.XGBoost.and.Scikit-learn.4113.Copyleaks.Report.pdf)

**Comment by Bravinsimiyu** [comment]
let me work on the changes ill let you know @srishilesh 
Thanks

**Comment by srishilesh** [comment]
@Bravinsimiyu 

Please ensure that you make changes acc. to the plagiarism check as well.

**Comment by Bravinsimiyu** [comment]
Hello @srishilesh, I also mad the changes according to the plagiarism check

**Review by srishilesh — APPROVED** [review]
@Bravinsimiyu 

Great article. It was very beginner friendly.

Moving for the final review.

Thanks.
